### PR TITLE
Create the inventory class.

### PR DESCRIPTION
**Pull request by p2hang**
The base inventory object provided three methods
get_product, get_all, put_product, delete
I think we should be able to get all the REST api implementation done with these actions
(e.g, update is basically get and modify and put.)

dict inventory is a subclass of the base inventory.

### CODE REVIEW

File: dict_inventory.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  def get_all(self):
+    result = {}
+    for k in self.products:
+      result[k] = self.products[k].__str__()
+    return result
+
+
+  def put_product(self, id, info):
+    self.products[id] = Product(id, info['location_id'], info['used'],
```
Comment (pinocchiochang) [inline dict_inventory.py:34]: How about if no this product's id?

File: dict_inventory.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                                info['new'], info['open_box'], info['restock_level'])
+    return True
+
+
+  def delete_product(self, product_id):
+    del self.products[product_id]
```
Comment (pinocchiochang) [inline dict_inventory.py]: Same as above, no this id in inventory?

**Review by pinocchiochang — COMMENTED** [review]
Comment in the codes.

File: dict_inventory.py (hunk 1 above)
Comment (p2hang) [reply dict_inventory.py:34]: If there is no such product id 
It should mean that the method is called by the create_product API (first time put into database)

File: dict_inventory.py (hunk 2 above)
Comment (p2hang) [reply dict_inventory.py]: Good catch, Thanks! 
And fixed.
Actually del will cast keyError for product id not found
but we should explicitly catch the error and print with custom error message.

File: dict_inventory.py (hunk 1 above)
Comment (pinocchiochang) [reply dict_inventory.py:34]: great

**Review by pinocchiochang — APPROVED** [review]
Approved.

**Review by omersolmazer — APPROVED** [review]
Approved!

**Pull request merged**